crates/pessimistic-proof/src/local_state.rs

```diff
@@ -54,7 +57,31 @@ impl LocalNetworkState {
         let mut new_balances: BTreeMap<_, _> =
             multi_batch_header.balances_proofs.iter().map(|(k, v)| (*k, v.0)).collect();
 
-        // TODO: check batch_header.imported_exits_root
+        // Check batch_header.imported_exits_root
+        let imported_exits_root = commit_imported_bridge_exits(
+            multi_batch_header.imported_bridge_exits.iter().map(|(exit, _)| exit),
+        );
+        if let Some(batch_imported_exits_root) = multi_batch_header.imported_exits_root {
+            if imported_exits_root != batch_imported_exits_root {
+                return Err(ProofError::InvalidImportedExitsRoot);
+            }
+        } else if !multi_batch_header.imported_bridge_exits.is_empty() {
+            return Err(ProofError::InvalidImportedExitsRoot);
+        }
+
+        let exits_hash = commit_bridge_exits(multi_batch_header.bridge_exits.iter());
+        let combined_hash =
+            keccak256_combine([exits_hash.as_slice(), imported_exits_root.as_slice()]);
```

Comment:
Minor: This bit of code calculating the signature message hash could be a `MultiBatchHeader` method. I suspect it might be useful for signing too.

Reply:
I think `MultiBatchHeader` is going to change quite a bit in the near future, so maybe we can add this method when the struct is stabilized.